### PR TITLE
Fix typo with mentions highlighting; simplify autocorrect logic; paste with attributes; warnings as errors

### PR DESCRIPTION
**Pull request by bkoatz**
1) Small typo for mentions highlighting fixed
2) Simplified autocorrection on/off logic in V2, turning it on when accessory view is present, and off when it disappears. This fixes an issue with inconsistent autocorrection when mentioning
3) Simplified autocorrection/autocapitalization/spellcheck on/off logic, to avoid 'cycling' as in v1, which led to comment reply box not showing on feed detail
4) Manually override paste events so that we can add attribution to the pasted text
5) Treat warnings as errors in line with voyager
6) Test all ^

### CODE REVIEW

File: Hakawai/Mentions/HKWMentionsPluginV2.m [hunk 1]
```diff
@@ -1006,6 +1007,26 @@ - (BOOL)textView:(__unused UITextView *)textView shouldChangeTextInRange:(NSRang
...
+                                                   attributes:[self defaultTextAttributes]];
+        }];
+        parentTextView.selectedRange = NSMakeRange(range.location + [text length], 0);
+        returnValue = NO;
+
+        // Manually notify external delegate that the textView changed
+        id<HKWTextViewDelegate> externalDelegate = parentTextView.externalDelegate;
+        if ([externalDelegate respondsToSelector:@selector(textViewDidChange:)]) {
+            [externalDelegate textViewDidChange:parentTextView];
+        }
```
Comment (li-jychen) [inline Hakawai/Mentions/HKWMentionsPluginV2.m:1027]: why is this only needed for pasting

File: Hakawai/Mentions/HKWMentionsPluginV2.m [hunk 2]
```diff
@@ -1006,6 +1007,26 @@ - (BOOL)textView:(__unused UITextView *)textView shouldChangeTextInRange:(NSRang
         [self bleachMentionsIntersectingWithRange:range];
     }
 
+    // If we just did a non-programmatic paste, insert the text directly and add attributes
+    // This is needed because rich text formatting is off, which strips attribution from pasting, which we want to maintain.
+    // Once (if ever) we turn rich text formatting on, we can remove this.
```
Comment (li-jychen) [inline Hakawai/Mentions/HKWMentionsPluginV2.m:1012]: wasn't the issue only exist for certain pasting flow?  in other cases for pasting, is rich text formatting on

File: Hakawai/Core/HKWControlFlowPluginProtocols.h [hunk 3]
```diff
@@ -49,8 +49,12 @@
 
 /*!
  If available, this method is called when the text view is about to paste text
+
+ @param textView Text view for paste
+ @param range Range for paste
+ @param programmatic  Whether the pasting is programmatic or not
  */
-- (void)textView:(UITextView *)textView willPasteTextInRange:(NSRange)range;
+- (void)textView:(UITextView *)textView willPasteTextInRange:(NSRange)range programmatic:(BOOL)programmatic;
```
[the remark this answers is not in the document]
Comment (li-jychen) [reply Hakawai/Core/HKWControlFlowPluginProtocols.h:57]: going to have a follow up PR to revisit existing pasting logic to see if we can call `[super paste]`first and then reapply attribute to simplify the logic.

File: Hakawai/Mentions/HKWMentionsPluginV2.m (hunk 1 above)
Comment (bkoatz) [reply Hakawai/Mentions/HKWMentionsPluginV2.m:1027]: explained on slack about that it's about overriding the shouldChangeText behavior

**Review by li-jychen — APPROVED** [review]

**Pull request merged**